src/BuffLog/BuffLog.php

```diff
@@ -47,27 +55,50 @@ public static function __callStatic($methodName, $args)
 
                 if (in_array($methodName, self::$logOutputMethods)) {
 
-                    // @TODO: need to make sure we "output" only the correct level of log
-                    //    old version looked like:
-                    //     self::setVerbosity();
-                    //     if (self::$currentVerbosity > Logger::WARNING) {
-                    //         return;
-                    //     }
+                    if (! self::shouldWriteLogs($methodName)) {
+                        return;
+                    }
 
                     self::enrichLog();
                 }
                 // Where the magic happen. We "proxy" functions name with arguments to the Monolog instance
                 return call_user_func_array(array(self::getLogger(), $methodName), $args);
 
             } else {
-
                 error_log("BuffLog::$methodName() is not supported yet. Add it to the BuffLog whitelist to allow it");
             }
         } else {
             error_log("BuffLog::$methodName() does not exist");
         }
     }
 
+    private static function setVerbosityLevel()
+    {
+        $logLevelFromEnv = getenv(self::$logLevelEnvVar);
+        $monologLevels = self::getLogger()->getLevels();
```

Comment:
We could slide this two lines out and pass `$logLevel` to the `setVerbosityLevel` as an argument. That way this method becomes a setter that could be used to configure the verbosity level in _user land._ I don't think that's going to happen very often, though.